### DISCUSSION TITLE
What is the naming rules of the file?

### DISCUSSION THREAD

**Discussion by feiyihx** [stashapp/stash#4743] [Q&A] [answered]
So my files are named under the following rules:
serialNumber description.performer.fileExtension

e.g. ABW-358C sex tutorial.涼森れむ.mp4

However, stash can't scrape any information about the video, only generates the cover picture.

I wonder if anyone can tell me what kind of naming rules I should use.

Thanks a lot!

**Comment by HereIam80** [comment, marked as answer]
That is easy to answer as stash doesn't care about file names at all. It is explicitly built to work with any file name or folder structure.
Also just a scan and generate task never add any metadata to your scenes. It just does what the description of the task says and nothing more.

The magic behind stash is the use of Phashes, which can recognize scenes based upon the contents of the scene and compare that to a community maintained database.

My advice is reading through this: https://docs.stashapp.cc/beginner-guides/guide-to-scraping/
However I don't think JAV scenes are on stashDB and you would need TPDB as a source for that.

If you really wanted to use your file names you can use the file name parser under settings => tools or maybe auto-tag but I never used those tools as both of those have limitations and are not really needed and many people using stash would advice against using those.

**Comment by feiyihx** [comment]
> That is easy to answer as stash doesn't care about file names at all. It is explicitly built to work with any file name or folder structure. Also just a scan and generate task never add any metadata to your scenes. It just does what the description of the task says and nothing more.
> 
> The magic behind stash is the use of Phashes, which can recognize scenes based upon the contents of the scene and compare that to a community maintained database.
> 
> My advice is reading through this: https://docs.stashapp.cc/beginner-guides/guide-to-scraping/ However I don't think JAV scenes are on stashDB and you would need TPDB as a source for that.
> 
> If you really wanted to use your file names you can use the file name parser under settings => tools or maybe auto-tag but I never used those tools as both of those have limitations and are not really needed and many people using stash would advice against using those.

Thanks a lot for the reply. TPDB doesn't work either, but I got the JAV library works. Thanks again for pointing me to the guides.
